common/lib/dependabot/errors.rb

```diff
@@ -617,4 +666,88 @@ class AllVersionsIgnored < DependabotError; end
 
   # Raised by FileParser if processing may execute external code in the update context
   class UnexpectedExternalCode < DependabotError; end
+
+  class RequiredVersionIsNotSatisfied < TypedDependabotError
+    sig { params(message: T.any(T.nilable(String), MatchData)).void }
+    def initialize(message = nil)
+      super("required_version_not_satisfied", message)
+    end
+  end
+
+  class AutoMergeParseFailure < TypedDependabotError
```

Comment:
I don't think we should put ecosystem-specific code into `common`. I see there's already a Go related error in this file, but it would make more sense to put these new errors into `npm_and_yarn/lib/errors.rb` if they only apply there.

Reply:
I created this error as not for eco-system but any eco-system can use this error to throw specific errors. I though ` RequiredVersionIsNotSatisfied` can be a global error. That's why I added to here.